### ISSUE TITLE
Custom SQL-Reports

### ISSUE THREAD

**Issue by Louis-Kre** [snipe/snipe-it#5975]
**Server (please complete the following information):**
 - Snipe-IT Version : 4.5.
 - OS: Ubuntu 16.04
 - Web Server: Apache2
 - PHP Version: 7.0

**Is your feature request related to a problem? Please describe.**
We need special reports for some categories. For example an report for all mobile devies and another one for all desktops. Every report has a different structure other columns.
Currently we have to download all columns that we need and import this in Excel, where we musst filter and sort everything. It is not so cool to download more than 400 rows, but you need only 20 rows...

**Describe the solution you'd like**
It would be cool, when you can register an custom view in the system and could show them in the frontend.
If you want to open, the system selects all columns from this view and display them in the frontend.

So we could create an SQL-View and could use them every time without export to excel. 

**Describe alternatives you've considered**
The next step could be filtering in the frontend or manage rights.

**Additional context**

Thank you, I'l looking forward for an implementation.

**Comment by snipe** [comment]
Can you not just use the Categories view and export from there? For example:
https://demo.snipeitapp.com/categories/1

Click on the export button and whatever is currently displayed on screen gets exported to a file.

**Comment by snipe** [comment]
(Additionally, you can select the category from the Custom Asset Report - https://demo.snipeitapp.com/reports/custom )

**Comment by Louis-Kre** [comment]
Yes thats correct, but than you get one file with the current status.
And for some reports we want to combine information that are not possible with this default layout.
For example: I want a table that contains all companies and all departments and a column, which contains the amount of laptops in this departments and so on.

We need a lot of such statistics for the finance department and for our it-statistics.

**Comment by snipe** [comment]
You would have to use the API to generate that information. 

The relationships between the different items in the system are pretty complicated, so I'm not sure arbitrary reports like what you're describing would be feasible.

**Comment by Louis-Kre** [comment]
I created some SQL-Views to combine this information.
For example, get all checkout assets and select the location (from the user or the checkout location hinself) and combine them with parent-location, like a department.
And know I could build an Excel-File that select all rows from the view an display them in the file.

But it would be nicer to have this reports in the snipe-it frontend.

I know custom reports could be very complicated for some users, but if you know how its work, you could build some helpfull reports.

**Comment by huangshifang** [comment]
SELECT departments.name as 部门名称, sum(departments.name='财务') as 财务, sum(departments.name='技术') as 技术, sum(departments.name='品管部') as 品管部 FROM assets,companies,users,departments WHERE assets.company_id=companies.id AND assets.assigned_to=users.id and department_id=departments.id 

I have the same request ,but I login to my mysql server and run the SQL of above in snipeit database,

**Comment by snipe** [comment]
> I know custom reports could be very complicated for some users, but if you know how its work, you could build some helpfull reports.

We know how it works - we built it. My point is that arbitrary SQL reports aren't going to be possible.

**Comment by Louis-Kre** [comment]
Is it not possible to create a "select * from view_xxx" and display this table in the frontend?
So we could generate our own SQL-Reports and share them in the frontend?

Thanks for your support.
